src/common/Constants.js

```diff
@@ -86,6 +85,26 @@ define(['common/core/constants', 'common/storage/constants'], function (CORE, ST
             RENAME_META_POINTER_TARGET: 'renameMetaPointerTarget',
             CHANGE_ASPECT_META: 'changeAspectMeta',
             REMOVE_META_RULE: 'removeMetaRule'
+        },
+
+        MULTILINE_MODE: {
+            generic: 'generic',
+            HTML: 'HTML',
+            'C-LIKE': 'C-LIKE',
+            CSS: 'CSS',
+            Erlang: 'Erlang',
+            Java: 'Java',
+            JavaScript: 'JavaScript',
+            Markdown: 'Markdown',
+            Mathematica: 'Mathematica',
+            Modelica: 'Modelica',
+            LaTeX: 'LaTeX',
+            Lua: 'Lua',
+            SQL: 'SQL',
+            Python: 'Python',
+            TTCN: 'TTCN',
+            YAML: 'YAML'
```

Comment:
I would have all of these in all lower-cases. Also breakdown `C-LIKE` to `c` and `cpp` (they can still map to codemirrors c-like). If `csharp` is highlighted in `clike` too..